packages/amplify-codegen/tests/commands/models.test.js

```diff
@@ -71,43 +72,148 @@ describe('command - models', () => {
+                    providerPlugin: 'awscloudformation',
+                    resourceName: MOCK_PROJECT_NAME
+                  }
+              ]
+          }
+      );
+    MOCK_CONTEXT.amplify.executeProviderUtils.mockReturnValue([]);
+    MOCK_CONTEXT.amplify.pathManager.getBackendDirPath.mockReturnValue(MOCK_BACKEND_DIRECTORY);
+    graphqlCodegen.codegen.mockReturnValue(MOCK_GENERATED_CODE);
   });
+
+  for (const frontend in OUTPUT_PATHS) {
+    it(frontend + ': Should clear the output directory before generation if cleanGeneratedModelsDirectory is true', async () => {
+      MOCK_CONTEXT.amplify.getProjectConfig.mockReturnValue({frontend: frontend});
+      require('amplify-cli-core').FeatureFlags.getBoolean.mockImplementation((name, defaultValue) => {
+          if (name === 'codegen.useappsyncmodelgenplugin' ||
+              name === 'codegen.cleanGeneratedModelsDirectory') {
+            return true;
+          }
+      });
+
+      const outputDirectory = path.join(MOCK_PROJECT_ROOT, OUTPUT_PATHS[frontend]);
+      const preExistingFilePath = path.join(outputDirectory, MOCK_PRE_EXISTING_FILE);
+      createMockFS(outputDirectory);
+
+      // assert output directory has a mock file to be cleared
+      expect(fs.readdirSync(outputDirectory).length).toEqual(1);
+      expect(fs.existsSync(preExistingFilePath));
+
+      await generateModels(MOCK_CONTEXT);
+
+      // assert model generation succeeds
+      expect(graphqlCodegen.codegen).toBeCalled();
+
+      // assert that codegen generated in correct place
+      expect(fs.readdirSync(outputDirectory).length).toBeGreaterThan(0);
+
+      // assert that the pre-existing file is deleted
+      expect(fs.existsSync(preExistingFilePath)).toBeFalsy;
+    });
+
+    it(frontend + ': Should not clear the output directory before generation if cleanGeneratedModelsDirectory is false', async () => {
+      MOCK_CONTEXT.amplify.getProjectConfig.mockReturnValue({frontend: frontend});
+      require('amplify-cli-core').FeatureFlags.getBoolean.mockImplementation((name, defaultValue) => {
+          if (name === 'codegen.useappsyncmodelgenplugin') {
+            return true;
+          }
+          if (name === 'codegen.cleanGeneratedModelsDirectory') {
+              return false;
+          }
+      });
+
+      const outputDirectory = path.join(MOCK_PROJECT_ROOT, OUTPUT_PATHS[frontend]);
+      const preExistingFilePath = path.join(outputDirectory, MOCK_PRE_EXISTING_FILE);
+      createMockFS(outputDirectory);
+
+      // assert output directory has a mock file to be cleared
+      expect(fs.readdirSync(outputDirectory).length).toEqual(1);
+      expect(fs.existsSync(preExistingFilePath));
+
+      await generateModels(MOCK_CONTEXT);
+
+      // assert model generation succeeds
+      expect(graphqlCodegen.codegen).toBeCalled();
+
+      // assert that codegen generated in correct place
+      expect(fs.readdirSync(outputDirectory).length).toBeGreaterThan(1);
```

Comment:
The previous test set the value to 0 but this is 1 instead. Is this intended change?

Reply:
This one tests `Should not clear the output directory before generation if cleanGeneratedModelsDirectory is false` and hence the pre-existing file should not be deleted.